pkg/keyspace/tso_keyspace_group.go

```diff
@@ -156,6 +172,31 @@ func (m *GroupManager) Close() {
 	m.wg.Wait()
 }
 
+// patrolKeyspaceAssignment is used to patrol all keyspaces and assign them to the keyspace groups.
+func (m *GroupManager) patrolKeyspaceAssignment() (err error) {
+	m.patrolKeyspaceAssignmentOnce.Do(func() {
+		var keyspaces []*keyspacepb.KeyspaceMeta
+		keyspaces, err = m.store.LoadRangeKeyspace(utils.DefaultKeyspaceID, 0)
+		if err != nil {
+			return
+		}
+		config, err := m.GetKeyspaceConfigByKind(endpoint.Basic)
+		if err != nil {
+			return
+		}
+		for _, ks := range keyspaces {
+			if ks == nil {
+				continue
+			}
+			err = m.UpdateKeyspaceForGroup(endpoint.Basic, config[TSOKeyspaceGroupIDKey], ks.GetId(), opAdd)
+			if err != nil {
```

Comment:
if meets err, then we will cannot continue to patrol other keyspaces?

Reply:
Nice catch. I think we can not use `sync.Once` here.

Reply:
Maybe we can put it into `allocNodesToAllKeyspaceGroups`, because they are similiar. If so, we need to change its name.